src/runtime_src/core/tools/xbmgmt2/SubCmdConfigure.cpp

```diff
@@ -0,0 +1,503 @@
+  auto value = xrt_core::query::data_retention::value_type(enable);
+  xrt_core::device_update<xrt_core::query::data_retention>(device, value);
+}
+
+void
+SubCmdConfigure::execute(const SubCmdOptions& _options) const
+{
+    XBU::verbose("SubCommand: configure");
+    // -- Retrieve and parse the subcommand options -----------------------------
+    // Common options
+    std::vector<std::string> devices;
+    std::string path;
+    std::string retention;
+    bool help = false;
+    // Hidden options
+    bool daemon = false;
+    std::string host;
+    std::string security;
+    std::string clk_scale;
+    std::string power_override;
+    std::string temp_override;
+    std::string cs_reset;
+    bool showx = false;
+
+    // Options previously under the load config command
+    po::options_description loadConfigOptions("Load Config Options");
+    loadConfigOptions.add_options()
+      ("input", boost::program_options::value<decltype(path)>(&path),"INI file with the memory configuration")
+    ;
+
+    // Options previously under the config command
+    po::options_description configOptions("Config Options");
+    configOptions.add_options()
+      ("retention", boost::program_options::value<decltype(retention)>(&retention),"Enables / Disables memory retention.  Valid values are: [ENABLE | DISABLE]")
+    ;
+
+    po::options_description commonOptions("Common Options");
+    commonOptions.add_options()
+        ("device,d", boost::program_options::value<decltype(devices)>(&devices)->multitoken(), "The Bus:Device.Function (e.g., 0000:d8:00.0) device of interest")
+        ("help,h", boost::program_options::bool_switch(&help), "Help to use this sub-command")
+    ;
+
+    commonOptions.add(loadConfigOptions);
+    commonOptions.add(configOptions);
+
+    // Hidden options previously under the config command
+    po::options_description configHiddenOptions("Hidden Options");
+    configHiddenOptions.add_options()
+        ("daemon", boost::program_options::bool_switch(&daemon), "Update the device daemon configuration")
+        ("host", boost::program_options::value<decltype(host)>(&host), "IP or hostname for device peer")
+        ("security", boost::program_options::value<decltype(security)>(&security), "Update the security level for the device")
+        ("runtime_clk_scale", boost::program_options::value<decltype(clk_scale)>(&clk_scale), "Enable/disable the device runtime clock scaling")
+        ("cs_threshold_power_override", boost::program_options::value<decltype(power_override)>(&power_override), "Update the power threshold in watts")
+        ("cs_threshold_temp_override", boost::program_options::value<decltype(temp_override)>(&temp_override), "Update the temperature threshold in celsius")
+        ("cs_reset", boost::program_options::value<decltype(cs_reset)>(&cs_reset), "Reset all scaling options")
+        ("showx", boost::program_options::bool_switch(&showx), "Display the device configuration settings")
```

Comment:
Hmm..the comments show that a description is needed, but in the code there is already one.  
Why the differences?

Reply:
Not sure what comments you are referring to. Do you mean code comments or the comments in the review? Shivangi left some comments about the description and they were fixed.